### PR TITLE
Update Kubernetes and OpenShift client versions and OpenShift client detection

### PR DESCRIPTION
**Pull request by hazim1093**

### CODE REVIEW

**Comment by stakater-user** [comment]
@hazim1093 Yikes! You better fix it before anyone else finds out! [Build 1](https://jenkins-delivery.stakater.com/job/Stakater/job/IngressMonitorController/job/PR-260/1/) has Failed!

**Comment by stakater-user** [comment]
@hazim1093 Yikes! You better fix it before anyone else finds out! [Build 2](https://jenkins-delivery.stakater.com/job/Stakater/job/IngressMonitorController/job/PR-260/2/) has Failed!

**Comment by stakater-user** [comment]
@hazim1093 Yikes! You better fix it before anyone else finds out! [Build 3](https://jenkins-delivery.stakater.com/job/Stakater/job/IngressMonitorController/job/PR-260/3/) has Failed!

**Comment by stakater-user** [comment]
@hazim1093 Image is available for testing. ``docker pull stakater/ingressmonitorcontroller:SNAPSHOT-PR-260-5``

**Comment by stakater-user** [comment]
@hazim1093 Image is available for testing. ``docker pull stakater/ingressmonitorcontroller:SNAPSHOT-PR-260-6``

**Comment by stakater-user** [comment]
@hazim1093 Image is available for testing. ``docker pull stakater/ingressmonitorcontroller:SNAPSHOT-PR-260-7``

**Comment by stakater-user** [comment]
@hazim1093 Yikes! You better fix it before anyone else finds out! [Build 8](https://jenkins-delivery.stakater.com/job/Stakater/job/IngressMonitorController/job/PR-260/8/) has Failed!

**Comment by stakater-user** [comment]
@hazim1093 Image is available for testing. ``docker pull stakater/ingressmonitorcontroller:SNAPSHOT-PR-260-9``

File: pkg/controller/controller_test.go [hunk 1]
```diff
@@ -734,11 +735,11 @@ func TestAddIngressWithAnnotationAssociatedWithServiceAndHasPodButNoProbesShould
...
 	}
 
-	controller.kubeClient.ExtensionsV1beta1().Ingresses(namespace).Delete(ingressName, &meta_v1.DeleteOptions{})
+	controller.kubeClient.ExtensionsV1beta1().Ingresses(namespace).Delete(context.TODO(), ingressName, meta_v1.DeleteOptions{})
 
-	controller.kubeClient.Core().Pods(namespace).Delete(podName, &meta_v1.DeleteOptions{})
+	controller.kubeClient.CoreV1().Pods(namespace).Delete(context.TODO(), podName, meta_v1.DeleteOptions{})
 
-	controller.kubeClient.Core().Services(namespace).Delete(serviceName, &meta_v1.DeleteOptions{})
+	controller.kubeClient.CoreV1().Services(namespace).Delete(context.TODO(), podName, meta_v1.DeleteOptions{})
```
Comment (waseem-h) [inline pkg/controller/controller_test.go:742]: Shouldn't `podName` be `serviceName`??

File: pkg/controller/controller_test.go [hunk 2]
```diff
@@ -811,11 +812,11 @@ func TestAddIngressWithHealthAnnotationAssociatedWithServiceAndHasPodShouldCreat
...
```
Comment (waseem-h) [inline pkg/controller/controller_test.go:819]: Same here. podName should be serviceName

File: pkg/controller/controller_test.go [hunk 3]
```diff
@@ -880,9 +881,9 @@ func TestAddIngressWithAnnotationAssociatedWithServiceAndHasNoPodShouldCreateMon
 		}
 	}
 
-	controller.kubeClient.ExtensionsV1beta1().Ingresses(namespace).Delete(ingressName, &meta_v1.DeleteOptions{})
+	controller.kubeClient.ExtensionsV1beta1().Ingresses(namespace).Delete(context.TODO(), ingressName, meta_v1.DeleteOptions{})
 
-	controller.kubeClient.Core().Services(namespace).Delete(serviceName, &meta_v1.DeleteOptions{})
+	controller.kubeClient.CoreV1().Services(namespace).Delete(context.TODO(), podName, meta_v1.DeleteOptions{})
```
Comment (waseem-h) [inline pkg/controller/controller_test.go:886]: Here too. It should be serviceName

**Comment by stakater-user** [comment]
@hazim1093 Image is available for testing. ``docker pull stakater/ingressmonitorcontroller:SNAPSHOT-PR-260-10``

**Comment by stakater-user** [comment]
@hazim1093 Image is available for testing. ``docker pull stakater/ingressmonitorcontroller:SNAPSHOT-PR-260-11``

**Comment by waseem-h** [comment]
Also, please run go mod download and go mod tidy to get rid of indirect dependencies

**Comment by stakater-user** [comment]
@hazim1093 Image is available for testing. ``docker pull stakater/ingressmonitorcontroller:SNAPSHOT-PR-260-12``

**Comment by stakater-user** [comment]
@hazim1093 Image is available for testing. ``docker pull stakater/ingressmonitorcontroller:SNAPSHOT-PR-260-13``

**Pull request merged**